components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java

```diff
@@ -96,15 +106,16 @@ public DropboxFileUploadResult put(String localPath, String remotePath, DropboxU
         }
     }
 
-    private DropboxFileUploadResult putFile(String localPath, DropboxUploadMode mode, String dropboxPath, DbxEntry entry) throws DropboxException {
+    private DropboxFileUploadResult putFile(String localPath, DropboxUploadMode mode, String dropboxPath, UploadUploader entry) throws DropboxException {
         File fileLocalPath = new File(localPath);
-        //verify uploading of a single file
+        // verify uploading of a single file
         if (fileLocalPath.isFile()) {
-            //check if dropbox file exists
-            if (entry != null && !entry.isFile()) {
+            // check if dropbox file exists
```

Comment:
We don't know if the entry is a file or not, so we may have to remove this check

Reply:
Well, if we are removing this, we will need to remove the 'entry' checks, meaning remove these check code entirely ?

// check if dropbox file exists
            if (entry != null) {
                throw new DropboxException(dropboxPath + " exists on dropbox and is not a file!");
            }
            // in case the entry not exists on dropbox check if the filename
            // should be appended
            if (entry == null) {
                if (dropboxPath.endsWith(DropboxConstants.DROPBOX_FILE_SEPARATOR)) {
                    dropboxPath = dropboxPath + fileLocalPath.getName();
                }
            }